nautobot/ipam/signals.py

```diff
@@ -65,16 +67,33 @@ def ip_address_to_interface_pre_delete(instance, raw=False, **kwargs):
     # that is the primary_v{version} of the host machine.
 
     if getattr(instance, "interface"):
-        host = instance.interface.device
-        other_assignments_exist = (
-            IPAddressToInterface.objects.filter(interface__device=host, ip_address=instance.ip_address)
-            .exclude(id=instance.id)
-            .exists()
-        )
+        interface = instance.interface
+        host = interface.device or interface.module.device
```

Comment:
Could use `interface.parent` here as that has the built-in logic to return either `.module.device` or `.device`.

Reply:
awesome, did not know that, added/changed it!